tuf/repository_tool.py

```diff
@@ -1953,31 +1961,28 @@ def add_target(self, filepath, custom=None):
     # freedom to add targets and parent restrictions in any order, minimize the
     # number of times these checks are performed, and allow any role to
     # delegate trust of packages to this Targes role.
-    if os.path.isfile(filepath):
 
-      # Update the role's 'tuf.roledb.py' entry and avoid duplicates.  Make
-      # sure to exclude the path separator when calculating the length of the
-      # targets directory.
-      targets_directory_length = len(self._targets_directory) + 1
-      roleinfo = tuf.roledb.get_roleinfo(self._rolename, self._repository_name)
-      relative_path = filepath[targets_directory_length:].replace('\\', '/')
+    # Update the role's 'tuf.roledb.py' entry and avoid duplicates.  Make
+    # sure to exclude the path separator when calculating the length of the
+    # targets directory.
+    targets_directory_length = len(self._targets_directory) + 1
+    roleinfo = tuf.roledb.get_roleinfo(self._rolename, self._repository_name)
+    relative_path = filepath[targets_directory_length:].replace('\\', '/')
```

Comment:
Nit (can probably be ignored in this PR): It might be good to add (or use) a path canonicalization method instead of calling `replace('\\', '/')` throughout the codebase.

Reply:
Agreed, this would be good cleanup.

@lukpueh any thoughts on whether this belongs in securesystemslib or in tuf? Perhaps as part of a new utils module?

Reply:
Absolutely! https://github.com/theupdateframework/tuf/pull/1008#pullrequestreview-382660018 has some thoughts about path normalization. The TLDR is, if the caller passes a *bad* path,
- either fail,
- or warn and normalize

In both cases this should happen once at the boundary where a path is added to the system, and thus remove the need for many `replace('\\', '/')` throughout the code.